salt/utils/jid.py

```diff
@@ -116,7 +114,7 @@ def jid_dir(jid, job_dir=None, hash_type='sha256'):
     Return the jid_dir for the given job id
     '''
     if not isinstance(jid, six.string_types):
-        jid = str(jid)
+        jid = six.text_type(jid)
     if six.PY3:
         jid = jid.encode('utf-8')
```

Comment:
Maybe these few lines should be replaced with   jid = salt.utils.to_bytes(jid)

Reply:
hashlib only requires bytes on Python 3, so this is not necessary.